vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -78,69 +132,176 @@
                     console.warn("The locale is not supported, using default locale setting(en-US).");
                 }
 
-                let currentDate = false;
+                /** @type {DateHash} */
+                let currentDate;
+                /** @type {string} */
                 let inputValue = getInputValue();
-                if (datepicker.i18n.parseDate !== 'undefined' && inputValue) {
+
+                if (datepicker.i18n.parseDate && inputValue) {
                     /* get current date with old parsing */
                     currentDate = datepicker.i18n.parseDate(inputValue);
                 }
 
                 /* create test-string where to extract parsing regex */
-                let testDate = new Date(Date.UTC(datepicker.$connector.yearPart.initial, datepicker.$connector.monthPart.initial - 1, datepicker.$connector.dayPart.initial));
+                let testDate = new Date(Date.UTC(parseInt(yearPart.initial), parseInt(monthPart.initial) - 1, parseInt(dayPart.initial)));
```

Comment:
Added explicit casting from string to number here just to make the IDE hinting happy and not show any warnings now that there is type info with JSDoc. It was complaining about the year and day here but I also changed month for consistency.